### PR TITLE
Closes #342 | Implement dataloader for SeaEval

### LINKED ISSUE #342: Create dataset loader for SeaEval
Dataloader name: `seaeval/seaeval.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?seaeval

|     Dataset| seaeval |
|-------------|---|
| Description | SeaEval is a benchmark toolkit for evaluating multilingual LLMs. The full SeaEval benchmark contains 28 datasets covering 7 languages. SeaEval introduces 2 new datasets for cross-lingual consistency, each containing parallel questions for the 7 represented languages. SeaEval also introduces 4 new datasets for cultural reasoning (multiple choice Q & A) that are in English but focused on regions including Singapore and the Philipines. |
| Subsets | Cross-MMLU, Cross-LogiQA, SG-Eval, PH-Eval |
| Languages | ind, vie, zlm, fil |
| Tasks | Commonsense Reasoning, Question Answering |
| License | Creative Commons Attribution Non Commercial 4.0 (cc-by-nc-4.0) |
| Homepage | https://github.com/SeaEval/SeaEval |
| HF URL | https://huggingface.co/datasets/SeaEval/SeaEval_datasets |
| Paper URL | https://arxiv.org/pdf/2309.04766.pdf |

### PR DESCRIPTION
**Pull request by raileymontalan**
Closes #342

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/seaeval/seaeval.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by holylovenia** [comment]
Hi @raileymontalan, could you also please add the corresponding language IDs to the subset names?

Examples:
- `seaeval_cross_mmlu_ind_seacrowd_qa`
- `seaeval_cross_mmlu_vie_seacrowd_qa`
- `seaeval_cross_mmlu_zlm_seacrowd_qa`
- `seaeval_cross_mmlu_fil_seacrowd_qa`
- `seaeval_cross_logiqa_ind_seacrowd_qa`
- `seaeval_cross_logiqa_vie_seacrowd_qa`
- `seaeval_cross_logiqa_zlm_seacrowd_qa`
- `seaeval_cross_logiqa_fil_seacrowd_qa`
- `seaeval_sg_eval_eng_seacrowd_qa`
- `seaeval_ph_eval_eng_seacrowd_qa`

File: seacrowd/sea_datasets/seaeval/seaeval.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        elif self.config.schema == "seacrowd_qa":
+            for row in examples:
+                x = {
+                    "id": idx,
+                    "question_id": row["id"],
+                    "document_id": row["id"],
+                    "question": row["question"],
+                    "type": "multiple_choice",
+                    "choices": row["choices"],
+                    "context": row["context"] if "context" in row else None,
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/seaeval/seaeval.py:231]: How about:

```
row.get("context")
```

For this type of logic? It returns `None` if the key itself doesn't exist.

**Review by ljvmiranda921 — COMMENTED** [review]
Dataloader works well! Just have some small nits on code (see comments). Thanks for the PR!

File: seacrowd/sea_datasets/seaeval/seaeval.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        data = {key: dl_manager.download_and_extract(value) for key, value in _URLS.items()}
+
+        paths = {}
+        file = self.config.subset_id.split("_")
+        file = "_".join(file[1:3])
+        paths[self.config.subset_id] = data[file]
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
```
Comment (SamuelCahyawijaya) [inline seacrowd/sea_datasets/seaeval/seaeval.py:161]: Since SeaEval is used for evaluation only, let's.update the split into `TEST` instead of `TRAIN`

**Review by SamuelCahyawijaya — CHANGES_REQUESTED** [review]
@raileymontalan, thank you for the contribution! The dataset works well, just a small nit on the dataset split.

File: seacrowd/sea_datasets/seaeval/seaeval.py (hunk 1 above)
Comment (raileymontalan) [reply seacrowd/sea_datasets/seaeval/seaeval.py:231]: The `row` object is of type `numpy.recarray` and doesn't have a `.get()` method. There doesn't seem to be a built-in method for `nump.recarray` that has the same behavior as `.get()` for Python `dict` types.

**Comment by raileymontalan** [comment]
@SamuelCahyawijaya changed the split to TEST. Thanks!

**Review by ljvmiranda921 — APPROVED** [review]

**Review by SamuelCahyawijaya — APPROVED** [review]
LGTM!

**Pull request merged**